src/Npgsql/KerberosUsernameProvider.cs

```diff
@@ -18,75 +15,25 @@ sealed class KerberosUsernameProvider
     static string? _principalWithRealm;
     static string? _principalWithoutRealm;
 
-    internal static ValueTask<string?> GetUsernameAsync(bool includeRealm, ILogger connectionLogger, bool async, CancellationToken cancellationToken)
+    internal static async ValueTask<string?> GetUsernameAsync(bool includeRealm, ILogger connectionLogger, bool async, CancellationToken cancellationToken)
     {
         if (_performedDetection)
-            return new(includeRealm ? _principalWithRealm : _principalWithoutRealm);
-        var klistPath = FindInPath("klist");
-        if (klistPath == null)
-        {
-            connectionLogger.LogDebug("klist not found in PATH, skipping Kerberos username detection");
-            return new((string?)null);
-        }
-        var processStartInfo = new ProcessStartInfo
-        {
-            FileName = klistPath,
-            RedirectStandardOutput = true,
-            RedirectStandardError = true,
-            UseShellExecute = false
-        };
+            return includeRealm ? _principalWithRealm : _principalWithoutRealm;
 
```

Comment:
Maybe leave the part above here as sync function and move the part below to a local async function (as it was before).
That way access to the cached values works without async state machine.

Reply:
Fixed.